runtime-modules/storage/src/lib.rs

```diff
@@ -568,8 +531,50 @@ impl<StorageBucketId: Ord, DistributionBucketId: Ord, Balance>
             }
         }
     }
+
+    fn with_storage_buckets(self, stored_by: BTreeSet<StorageBucketId>) -> Self {
+        Self { stored_by, ..self }
+    }
+
+    fn with_distribution_buckets(self, distributed_by: BTreeSet<DistributionBucketId>) -> Self {
+        Self {
+            distributed_by,
+            ..self
+        }
+    }
+
+    fn with_prize(self, deletion_prize: Balance) -> Self {
+        // Option<T> defaults to None
+        Self {
+            deletion_prize: Some(deletion_prize),
+            ..self
+        }
+    }
 }
 
+// Helper enum for performing bag operation: no default since it is non trivial
+#[derive(Clone, PartialEq, Eq, Debug)]
+enum BagOperationParamsTypes<Balance: Unsigned, ObjectId: Clone> {
+    Update(Vec<DataObjectCreationParameters>, BTreeSet<ObjectId>),
```

Comment:
Please, comment the options and parameters. Consider switching to the named parameters for enum options.

Reply:
:+1: